### ISSUE TITLE
train command?

### ISSUE THREAD

**Issue by designerZhou** [mmathew23/LAG-Pytorch#1]
thanks for your work ,i want use your git to train my dataset, may i get the command of training ?

**Comment by mmathew23** [comment]
Hi, thanks for checking out this project. It's still a work in progress. Right now, there are stability issues with training and I can't get the model to train properly. I haven't narrowed down where my bug is yet. If you'd like to play around you can run python train.py --root_path /path_to_data/ --batch_size bs, all the command line options are the same as the arguments to the train function defined in train.py.

**Comment by Athenanna** [comment]
hi, thanks for your work, and i trained with my own data, while there are stability issues, losses increases rapidly after step-15k, Loss_D/dfake grows to 1.8e12, Loss_G/gfake  to -1.8e12. have you found the reason or solved it?

**Comment by mmathew23** [comment]
I've been sidetracked on other projects, so unfortunately, no, I haven't figured out the issues yet. I've departed from the exact paper implementation a bit, so it's not as straightforward to troubleshoot. Most notably I'm not using the progressive growing training strategy and instead opt for something more like stylegan2's strategy. If you don't mind sharing, what hyper parameters did you use for training? What kind of data are you using?

**Comment by Athenanna** [comment]
i used CelebA-HQ(resize to 256x256), sr16x (from 16x16 to 256x256), and use multiGPU training(nn.DataParallel), so the main differences are:
**1. nn.DataParallel()
2.  Discriminator use GPU (not .cuda('cpu')), due to nn.DataParallel()
3. upsample scale: 16X 
4. rewrite "sres"(Generator() in your projects) ：**

`class SRES(nn.Module):
    def __init__(self, max_filters=256, min_filters=64, noise_dim=64, blocks=8):
        super(SRES, self).__init__()
        self.noise_dim = noise_dim
        self.conv1 =  nn.Conv2d(3+noise_dim, max_filters, 3, padding=1)
        self.resblock = nn.Sequential(*[ResidualBlock(max_filters) for block in range(blocks)])

        self.relu = nn.ReLU()
        self.lrelu = nn.LeakyReLU(negative_slope=0.2)

        self.upsample = nn.UpsamplingNearest2d(scale_factor=2)

        self.upsample_layer1 = UpsampleBlock(256,128)
        self.upsample_layer2 = UpsampleBlock(128,64)
        self.upsample_layer3 = UpsampleBlock(64,64)
        self.upsample_layer4 = UpsampleBlock(64,64)

        self.rgb1 = nn.Conv2d(128, 3, 3, padding=1)
        self.rgb2 = nn.Conv2d(64, 3, 3, padding=1)
        self.rgb3 = nn.Conv2d(64, 3, 3, padding=1)
        self.rgb4 = nn.Conv2d(64, 3, 3, padding=1)


    def forward(self, x, noise_scale=1, eps=None):
        b, c, h, w = x.shape
        if eps is None:
            eps = torch.randn(b, self.noise_dim, h, w)
        eps *= noise_scale
        eps = eps.cuda()


        x = self.conv1(torch.cat([x, eps], dim=1))

        x = self.resblock(x)

        x1 = self.upsample_layer1(x)
        x2 = self.upsample_layer2(x1)
        x3 = self.upsample_layer3(x2)
        x4 = self.upsample_layer4(x3)

        x1_rgb = self.rgb1(x1)
        x2_rgb = self.rgb2(x2) + self.upsample(x1_rgb)
        x3_rgb = self.rgb3(x3) + self.upsample(x2_rgb)
        x4_rgb = self.rgb4(x4) + self.upsample(x3_rgb)

        return x4_rgb`

**And hyper parameters:**

root_path='CelebA-HQ'
    batch_size=16
    epochs=100
    lr=.0001
    max_filters=256
    min_filters=64
    upsample_layers=4
    noise_dim=64
    blocks=8
    device_id=[1,2,3,4]
    image_size=256
    num_workers=0
    wass_target=1
    mse_weight=10
    ttur=4
    models_dir='./models/'
    results_dir='./results/'
    log_dir='./log/'
    model_name_prefix='test'
    save_every=1000
    batch_shuffle = True

**Comment by mmathew23** [comment]
Thanks for providing this info! I intended to add DataParallel so I will add that.  I'll also carve out some time and figure out what's wrong with the stability.

**Comment by mmathew23** [comment]
I've fixed some of the stability problems and added DataParallel support. Still some work to do as training error jumps significantly in the beginning and then starts improving. I don't think that should happen, but I've been able to get it to train on my data. I may tweak the gan architecture a bit to see if more improvements can be made.
